cmd/builder/internal/builder/main.go

```diff
@@ -167,7 +168,7 @@ func GetModules(cfg Config) error {
 	if !ok {
 		return fmt.Errorf("core collector %w: '%s'. %s", ErrDepNotFound, corePath, skipStrictMsg)
 	}
-	if coreDepVersion != coreVersion {
+	if semver.MajorMinor(coreDepVersion) != semver.MajorMinor(coreVersion) {
```

Comment:
I think this should be alright, but could we use `semver.Compare` here?

Reply:
`semver.Compare` includes the patch version in its comparison, but my goal is to allow patch versions to be different.